functions/compare_classification_facts.pp

```diff
@@ -0,0 +1,39 @@
+# @summary Compare untrusted values from the classification fact to their trusted values.
+#
+# Compare untrusted values from the classification fact to their trusted values.
+#
+# @example
+#   $classification_calculated_trusted = $parsed_classification + {
+#     hostname => $hostname,
+#   }
+#   $classification_fact_differences = classification::compare_classification_facts($classification_calculated_trusted)
+#
+# @param classification_calculated_trusted A hash of facts to validate
+# @return [Array] An array of strings documenting any facts that don't match their trusted values
+function classification::compare_classification_facts(Hash $classification_calculated_trusted) >> Array {
```

Comment:
You could combine this with `classification::compare_root_level_facts`. For example:

~~~ puppet
# I'm just shortening stuff for legibility here
function classification::compare_trusted(Hash $untrusted, Hash $trusted, Array $names) >> Array {
  $differences = delete_undef_values($names.map |$fact_name| {
    $untrusted_value = String($untrusted[$fact_name])
    $trusted_value = String($trusted[$fact_name])
    . . .
}


$root_level_fact_differences = classification::compare_trusted(
  $facts,
  $root_level_calculated_trusted,
  [
    'certname',
    'hostname',
    'fqdn',
    'domain',
  ])

$classification_fact_differences = classification::compare_trusted(
  $facts['classification'],
  $classification_calculated_trusted,
  [
    'hostname',
    'version',
    'group',
    'function',
    'number',
    'number_string',
    'context',
    'stage',
  ])
~~~

Reply:
done and thanks!